nova_vm/src/ecmascript/abstract_operations/testing_and_comparison.rs

```diff
@@ -277,3 +277,133 @@ pub(crate) fn is_less_than<const LEFT_FIRST: bool>(
         Ok(Some(rnx < rny))
     }
 }
+
+/// [7.2.14 IsLooselyEqual ( x, y )](https://tc39.es/ecma262/#sec-islooselyequal)
+///
+/// The abstract operation IsLooselyEqual takes arguments x (an ECMAScript
+/// language value) and y (an ECMAScript language value) and returns either a
+/// normal completion containing a Boolean or a throw completion. It provides
+/// the semantics for the == operator. It performs the following steps when
+/// called:
+pub(crate) fn is_loosely_equal(
+    agent: &mut Agent,
+    x: impl Into<Value> + Copy,
+    y: impl Into<Value> + Copy,
+) -> JsResult<bool> {
+    let (x, y) = (x.into(), y.into());
+
+    // 1. If Type(x) is Type(y), then
+    if is_same_type(x, y) {
+        // a. Return IsStrictlyEqual(x, y).
+        return Ok(is_strictly_equal(agent, x, y));
+    }
+
+    // 2. If x is null and y is undefined, return true.
+    if x.is_null() && y.is_undefined() {
+        return Ok(true);
+    }
+
+    // 3. If x is undefined and y is null, return true.
+    if x.is_undefined() && y.is_null() {
+        return Ok(true);
+    }
+
+    // 4. Perform the following steps:
+    // a. If x is an Object, x has an [[IsHTMLDDA]] internal slot, and y is either undefined or null, return true.
+    // b. If x is either undefined or null, y is an Object, and y has an [[IsHTMLDDA]] internal slot, return true.
+
+    // 5. If x is a Number and y is a String, return ! IsLooselyEqual(x, ! ToNumber(y)).
+    if x.is_number() && y.is_string() {
+        let y = to_number(agent, y)?;
```

Comment:
**issue:** Spec text says `! ToNumber(y)` so this should be `to_number(agent, y).unwrap()`.